hack/jenkins/e2e.sh

```diff
@@ -364,6 +364,32 @@ case ${JOB_NAME} in
     ADMISSION_CONTROL="NamespaceLifecycle,InitialResources,LimitRanger,SecurityContextDeny,ServiceAccount,ResourceQuota"
     ;;
 
+  # Runs only the ingress tests on GCE.
+  kubernetes-e2e-gce-ingress)
+    : ${E2E_CLUSTER_NAME:="jenkins-gce-e2e-ingress"}
+    : ${E2E_NETWORK:="e2e-ingress"}
+    : ${GINKGO_TEST_ARGS:="--ginkgo.focus=\[Feature:Ingress\]"}
+    : ${KUBE_GCE_INSTANCE_PREFIX:="e2e-ingress"}
+    : ${FAIL_ON_GCP_RESOURCE_LEAK:="true"}
+    # TODO: Move this into a different project. Currently, since this test
+    # shares resources with various other networking tests, so it's easier
+    # to zero in on the source of a leak if it's run in isolation.
+    : ${PROJECT:="kubernetes-flannel"}
+    ;;
+
+   # Runs only the ingress tests on GKE.
+   kubernetes-e2e-gke-ingress)
```

Comment:
You need to make a separate suite, `kubernetes-e2e-gke-ingress`, but mock the `kubernetes-e2e-gke`, since there are some other ENVs for that provider.

Also note you need to add jobs to `hack/jenkins/job-configs/kubernetes-e2e.yaml`; Jenkins configs are now source-controlled there.  cc @spxtr